src/angular-app/languageforge/lexicon/editor/editor.component.ts

```diff
@@ -369,6 +372,10 @@ export class LexiconEditorController implements angular.IController {
     // entry and is NOT going to a different entry (as is the case with editing another entry.
     let newEntryTempId: string;
 
+    // We might be saving, because the user is navigating away from this entry,
+    // in which case we don't want to lose the upload.
+    await this.recordingStateService.uploading$();
```

Comment:
@rmunn So, we either: (1) ditch the audio and hope as much gets saved as possible (2) wait for the audio at the risk of the save then never getting run at all or (3) put effort into configuring when we wait.

Reply:
It's worth mentioning that this is actually just a bonus. The `return` on 1141 is much more significant and effective.